components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -4540,6 +4580,255 @@ public void deleteApplicationByResourceId(String resourceId,
         }
     }
 
+    @Override
+    public List<ApplicationBasicInfo> getDiscoverableApplicationBasicInfo(int limit, int offset, String filter,
+                                                                          String sortOrder, String sortBy, String
+                                                                                  tenantDomain) throws
+            IdentityApplicationManagementException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Retrieving application basic information of discoverable applications for limit: " + limit +
+                    " offset: " + offset + " filter: " + filter + " sortOrder: " + sortOrder + " sortBy: " + sortBy +
+                    " in tenantDomain: " + tenantDomain);
+        }
+
+        validateForUnImplementedSortingAttributes(sortOrder, sortBy);
+        validateAttributesForPagination(offset, limit);
+
+        // TODO: 11/5/19 : Enforce a max limit
```

Comment:
Let's create a GitHub issue to track this if we are not fixing this in this iteration.

Reply:
Added https://github.com/wso2/product-is/issues/6802